test/strong_params/filter_test.exs

```diff
@@ -5,7 +5,7 @@ defmodule StrongParams.FilterTest do
   alias StrongParams.Filter
 
   describe "apply/3" do
-    test "filter and return required fields" do
+    test "filters and return required fields" do
```

Comment:
@rog404 que cenário vc tá pensando em cobrir. Pq no controller sempre vai vir como string. Não faz sentido suportar uma key com atom, pq o frontend nunca consegue passar atom-keys

Reply:
Opa @thiamsantos o caso que estamos fazendo um `required` dentro de um param que e `permitted`.

Por exemplo:

```ex
filter_for(:create,
    permitted: [
      {
      :fields,
       [
         [
           {:field_id, UUID},
           {:field_array, {:array, UUID}}
         ]
       ]
       },
      ]
    ],
    forbidden_params_err: true
  )
  ```
  
  Gostaríamos deixar os campos de `fields` obrigatório caso eles apareçam. 
  
  Estamos no nosso controller criando uma função pra isso:
  
  ```ex
    defp put_fields(%{fields: fields} = params) do
    map = %{"fields" => fields}

    with {:ok, %{"fields" => fields}} <-
           filter_params(map,
             required: [
               {:fields,
                [
                  [
                    {:field_id, UUID},
                    {:field_array, {:array, UUID}}
                  ]
                ]}
             ]
           ) do
      {:ok, Map.put(params, :fields, fields)}
    end
  end
  ```
  
  Atualmente, ela geraria um erro pois: O plug já transforma todos os fields para `atom key`.
  Ou seja, ao usar o `filter_params` ele vai falhar com status de `field_id is required` pois ele não encontra o  `%{"fields_id" => "xxx"}` e sim o `%{fields_id: "xxx"}`. 
  
  Faz sentido?